test/e2e/autoscaling/cluster_size_autoscaling.go

```diff
@@ -207,101 +207,106 @@ var _ = SIGDescribe("Cluster size autoscaling [Slow]", func() {
 	It("should increase cluster size if pending pods are small [Feature:ClusterSizeAutoscalingScaleUp]",
 		func() { simpleScaleUpTest(0) })
 
-	It("Should scale up GPU pool from 0 [Feature:ClusterSizeAutoscalingGpu]", func() {
-		framework.SkipUnlessProviderIs("gke")
+	supportedGpuTypes := []string{"nvidia-tesla-k80", "nvidia-tesla-v100", "nvidia-tesla-p100"}
```

Comment:
This is problematic. Currently all the projects have k80 and p100 quota but v100 quota is only available in one project. So, there is no project that can successfully run this.

Can you instead take the gpu type as argument that is passed by the test-infra job? Then we can have three jobs that each run the tests for different gpu type. This will also ensure that we can easily add new gpu types (just a test-infra config change instead of change in k/k) and run them at different frequency (run v100 tests less frequently as they are more expensive) etc.

Reply:
Thanks for comment.

Actually I thought about it. Note that the loop over the supported gpu types is only for generating testcases. As part of this PR the name of each testcase is changed so it contains `[GpuType:%s]` with `%s` being `nvidia-tesla-k80`, `nvidia-tesla-p100`, ... . This then testcases can be easily used for filtering out/split into multiple test jobs using `--ginkgo.focus`.

Reply:
Take a look at this PR please: https://github.com/kubernetes/test-infra/pull/8144

Reply:
What is the advantage of generating test cases like this instead of just passing the gpuType from test-infra?

With this approach, when adding a new GPU type you would still need two PRs, one in kubernetes/kubernetes and another in kubernetes/test-infra.

Reply:
> If set of tested GPU types depend on external parametrization we can easily skip coverage of GPU+CA interoperability just by forgetting to pass the list of accelerator types.

This can still happen. You can add a new gpu type here but forget to add it in test-infra and then it will not get tested. So, I don't agree this is more safe. It's just twice the work.

Reply:
It is mostly about having single ropository (kubernetes/kubernetes) being source of truth in the matter what tests are defined. 
But I will not fight for that.

We can change:
```golang
supportedGpuTypes := []string{"nvidia-tesla-k80", "nvidia-tesla-v100", "nvidia-tesla-p100"}
for _, gpuType := range supportedGpuTypes {
```

to sth like
```golang
gpuTypes := getTestedGpuTypesPassedViaTestJobParameters()
for _, gpuType := range gpuTypes {
```

Are you aware how `getTestedGpuTypesPassedViaTestJobParameters` should be implemented? We need to pass the information from `kubernetes_e2e.py` params to tests code. Is someone already doing that?

Btw. Maybe it is worthy to merge the new dashboards in the meantime as the old one is failing now due to timeout.

Reply:
Yeah, let's merge the new dashboards in the meantime.

I was thinking that we remove the for loop altogether and just have `gpuType` passed from test-infra.

You can do something like: https://github.com/kubernetes/test-infra/blob/02d3520f35c16efd494253ca074d9cad0826f84a/jobs/config.json#L7423 to pass the gpuType and then https://github.com/kubernetes/kubernetes/blob/99ebcd94c9404ae6c00498712b095ad132b8d447/test/e2e/scheduling/nvidia-gpus.go#L123 to read it in this file.

Reply:
Sorry for dealy. I was OOO couple of days.

> I was thinking that we remove the for loop altogether and just have gpuType passed from test-infra.

That would enforce separate test job for each gpu test. I guess this is fine. I am still not fine with having test definitions split between repositories.
I will change it though as you insist.

Reply:
Thank you!